### ISSUE TITLE
arrival time not updated when changing profile

### ISSUE THREAD

**Issue by FransM** [osmandapp/OsmAnd#15330]
### Description

I opened a gpx track that was made for bicycle. If I choose navigate and change the profile to e.g. car or pedestrian (walking) the estimated arrival time iin the right panel is not updated. (I'm not sure if this is a bug, but I expected an update).

(btw I also got some silly travel times when changing tracks. at one point it told me after changing a track to another one (where I entered midway) that I had 25 km to go and that it would take 7 min. I was by bike, but even with a car it would have required me driving at 200+ km/h). No idea how to reproduce this part)

### How to reproduce?

open a gpx file with a track, choose navigate, notice arrival time; change profile

### Your Environment
OsmAnd Version: 4,2,7
Android/iOS version: 12
Device model: Moto G pro

**Maps used (online or offline):**   
If you have an issue related to offline maps, tell us the exact name of the map file where the issue occurs and its edition date.

**Comment by vshcherb** [comment]
Not reproducible, in 4.2. That issue was fixed in 4.2 but it was present in 4.1

**Comment by FransM** [comment]
I just checked in 4.2.7 (fdroid version) and I think it is reproducible.
What I did was:I created a track consisting of 2 points. In options I then changed from straight line to bicycle.The route that was created was what I expected.
I saved the track.
Then I went to navigation, I opened the track, the profile with which it was opened was car (that is what was shown in the small icon left top below the directions).
Estimated arrival time was roughly what I expected for bicycle and more than I expected for car.
When I then change the profile to bicycle or pedestrian using the icon below the direction arrow the arrival time was not changed.
What I suspect is happening is that navigation uses whatever profile is in the gpx file to determine arrival time, not the actual profile.I'm not sure if that is what is supposed to happen but not what I expected.Maybe when a gpx track with bicycle profile in it is opened, the default profile when opening the track for navigation should result in changing the profile to bicycle.

Somewhat related: the route went through a road that is not car accessible.

    Op maandag 19 september 2022 om 15:51:53 CEST schreef vshcherb ***@***.***>:  
 
 


Closed #15330 as completed.
